processing/pom.xml

```diff
@@ -81,6 +81,11 @@
       <groupId>org.apache.commons</groupId>
       <artifactId>commons-compress</artifactId>
     </dependency>
+    <dependency>
+      <groupId>commons-codec</groupId>
+      <artifactId>commons-codec</artifactId>
+      <scope>runtime</scope>
```

Comment:
any reason we need to add this dependency to the processing pom?
It also seems odd we would only be declaring it a runtime dependency.

Reply:
it's added to processing as this is where commons-compress is used.
The dependency is listed as optional for commons-compress (https://mvnrepository.com/artifact/org.apache.commons/commons-compress/1.26.0) 
and is not needed compile time. The dependency is needed runtime (as illustrated by failing tests in https://github.com/apache/druid/pull/15932). If we add it in the compile scope we will need to  add the suppression of dependency analyze tool.

Reply:
I see, can we add a comment to explain the reason for making it runtime?

Let's also update the PR description to mention that 1.26 added this optional dependency and that it is required in our case to make tests pass.